src/components/KDropdown/KDropdown.vue

```diff
@@ -186,16 +186,21 @@ defineExpose({
   .dropdown-trigger {
     width: 100%;
   }
+}
+</style>
 
-  :deep(.popover.dropdown-popover > .popover-container) {
-    border: var(--kui-border-width-10, $kui-border-width-10) solid var(--kui-color-border, $kui-color-border);
-    border-radius: var(--kui-border-radius-30, $kui-border-radius-30);
-    padding: var(--kui-space-0, $kui-space-0);
-
-    ul {
-      margin: 0;
-      padding: var(--kui-space-20, $kui-space-20) 0;
-    }
+<style lang="scss">
+// We use a global style here because the popover may be teleported outside the component.
+// The selector might look unusual, but it’s intentional: keeping the same specificity
+// as before supporting teleportation ensures backward compatibility and avoids style conflicts.
+.k-dropdown-popover.dropdown-popover.popover > .popover-container {
+  border: var(--kui-border-width-10, $kui-border-width-10) solid var(--kui-color-border, $kui-color-border);
+  border-radius: var(--kui-border-radius-30, $kui-border-radius-30);
+  padding: var(--kui-space-0, $kui-space-0);
+
+  .dropdown-list {
+    margin: 0;
+    padding: var(--kui-space-20, $kui-space-20) 0;
```

Comment:
Could we utilize [global selectors](https://vuejs.org/api/sfc-css-features#global-selectors) here instead of needing the separate style block? (Applies to whole PR)

Reply:
`:global` needs to be applied to the whole selector which makes it not very friendly for nested usage.

Reply:
I was just suggesting moving your existing changes back into the single scoped style block as-is, but wrapped with :global